sdks/java/io/amazon-web-services2/src/main/java/org/apache/beam/sdk/io/aws2/kinesis/enhancedfanout/EFOShardSubscribersPool.java

```diff
@@ -96,10 +98,10 @@ class EFOShardSubscribersPool {
   /**
    * State map of currently active shards that can be checkpointed.
    *
-   * <p>This map may only be updated from within {@link #getNextRecord()} (and dependent {@link
-   * #onEventDone}).
+   * <p>This map may only be accessed and updated from within {@link #start}, {@link #getNextRecord}
+   * and dependent {@link #onEventDone} to prevent race conditions.
    */
-  private final Map<String, ShardState> state = new ConcurrentHashMap<>();
+  private final Map<String, ShardState> state = new HashMap<>();
```

Comment:
I suggest switching to a normal HashMap to make clear that concurrent access is not supported.